flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImpl.java

```diff
@@ -209,14 +228,22 @@ private static boolean isPartitionWithLocalResources(
                 .isPresent();
     }
 
-    private static Stream<ResultPartitionDeploymentDescriptor> excludePersistentPartitions(
+    private static Stream<ResultPartitionDeploymentDescriptor> excludePersistentAndLocalPartitions(
```

Comment:
I think we should not exclude partitions which occupy local resources. Because conceptually external partition can also occupy local resources.

Reply:
ok, though I don't find a scenario in which an completed external partition occupies tm resource. Note that if we keep excludePersistentPartitions, shufflemaster might be fed with unknown partitionid

Reply:
All partitions should have been registered to the `ShuffleMaster` no matter whether it occupies TM resources.
So I think `ShuffleMaster` would not see unknown partition ids.

Reply:
that's true